pint/registry.py

```diff
@@ -1400,6 +1485,7 @@ def _get_compatible_units(self, input_units, group_or_system):
         ret = super()._get_compatible_units(input_units, group_or_system)
 
         if self._active_ctx:
+            ret = ret.copy()  # Do not alter self._cache
```

Comment:
This line missing was the one and only reason why contexts required their own cache before!